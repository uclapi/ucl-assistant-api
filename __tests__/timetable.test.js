const axios = require(`axios`)

const {
  getPersonalWeekTimetable,
} = require(`../uclapi/timetable`)

const SAMPLE_TOKEN = `eySFsfef`
const SAMPLE_DATE = `2020-01-13`
const WRONG_DATE = `2020-01-14`
const SAMPLE_CLIENT_SECRET = `owjefjewfje`

jest.mock(`axios`)

describe(`timetable`, () => {
  beforeAll(() => {
    process.env.UCLAPI_CLIENT_SECRET = SAMPLE_CLIENT_SECRET
  })
  it(`should send a valid timetable request`, () => {
    getPersonalWeekTimetable(
      SAMPLE_TOKEN,
      SAMPLE_DATE,
    )
    expect(axios.get).toHaveBeenCalledTimes(5)
    expect(axios.get.mock.calls).toMatchSnapshot()
  })
  it(`should throw error if date is invalid`, () => {
    try {
      getPersonalWeekTimetable(
        SAMPLE_TOKEN,
        null,
      )
    } catch (error) {
      expect(error).toMatchSnapshot()
    }

    try {
      getPersonalWeekTimetable(
        SAMPLE_TOKEN,
        WRONG_DATE,
      )
    } catch (error) {
      expect(error).toMatchSnapshot()
    }
  })
})
