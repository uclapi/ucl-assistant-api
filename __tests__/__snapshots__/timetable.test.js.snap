// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timetable should send a valid timetable request 1`] = `
Array [
  Array [
    "https://uclapi.com/timetable/personal",
    Object {
      "params": Object {
        "client_secret": "owjefjewfje",
        "date_filter": "2020-01-13",
        "token": "eySFsfef",
      },
    },
  ],
  Array [
    "https://uclapi.com/timetable/personal",
    Object {
      "params": Object {
        "client_secret": "owjefjewfje",
        "date_filter": "2020-01-14",
        "token": "eySFsfef",
      },
    },
  ],
  Array [
    "https://uclapi.com/timetable/personal",
    Object {
      "params": Object {
        "client_secret": "owjefjewfje",
        "date_filter": "2020-01-15",
        "token": "eySFsfef",
      },
    },
  ],
  Array [
    "https://uclapi.com/timetable/personal",
    Object {
      "params": Object {
        "client_secret": "owjefjewfje",
        "date_filter": "2020-01-16",
        "token": "eySFsfef",
      },
    },
  ],
  Array [
    "https://uclapi.com/timetable/personal",
    Object {
      "params": Object {
        "client_secret": "owjefjewfje",
        "date_filter": "2020-01-17",
        "token": "eySFsfef",
      },
    },
  ],
]
`;

exports[`timetable should throw error if date is invalid 1`] = `[Error: Must specify date to retrieve weekly timetable]`;

exports[`timetable should throw error if date is invalid 2`] = `[Error: Date must be a Monday]`;
